printer_zpl2/models/printing_label_zpl2.py

```diff
@@ -123,10 +123,8 @@ def check_recursion(self):
 
     def _get_component_data(self, record, component, eval_args):
         if component.data_autofill:
-            data = component.autofill_data(record, eval_args)
-        else:
-            data = component.data
-        return safe_eval(str(data), eval_args) or ""
+            return component.autofill_data(record, eval_args)
```

Comment:
Why this change?

Reply:
Good day @etobella, thanks for your comment. We faced below error because `str(data)` convert `datetime` value to `datetime.datetime(2022, 10, 28, 8, 58, 26, 226475)` format. So instead of doing unnecessary work (convert to string then convert back to python), we just simply return the data. The `safe_eval` is still useful in the line after above line to convert string from db.
```
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/opt/OCA/report-print-send/tree/15.0/printer_zpl2/models/printing_label_zpl2.py", line 129, in _get_component_data
    return safe_eval(str(data), eval_args) or ""
  File "/opt/odoo/15.0/odoo/tools/safe_eval.py", line 348, in safe_eval
    raise ValueError('%s: "%s" while evaluating\n%r' % (ustr(type(e)), ustr(e), expr))
ValueError: <class 'NameError'>: "name 'datetime' is not defined" while evaluating
"{'id': 185, 'create_date': datetime.datetime(2022, 10, 28, 8, 58, 26, 226475)}"
```